src/codelabs/null-safety.md

```diff
@@ -0,0 +1,321 @@
+---
+title: Null safety codelab
+description: Learn about and practice writing null safe code in DartPad!
+js: [{url: 'https://dartpad.dev/inject_embed.dart.js', defer: true}]
+---
+{% assign useIframe = false -%}
+<?code-excerpt replace="/ *\/\/\s+ignore_for_file:[^\n]+\n//g; /(^|\n)*\/\/\s+ignore:[^\n]+\n/$1/g; /(\n[^\n]+) *\/\/\s+ignore:[^\n]+\n/$1\n/g"?>
+<?code-excerpt plaster="none"?>
+
```

Comment:
Throughout this codelab, please cross-link any concept (like "late" or "definite assignment") to the reference docs section for that concept in https://dart.dev/null-safety/understanding-null-safety